File: test/fixtures/definitions/sample.rs
// A Charge is settled here.
// Each of these lines is a comment of its own.
use std::fmt;

/// A doc comment is a line comment with a marker on it.
pub struct Charge {
    amount: u32,
}

impl Charge {
    // @contract Charge::settle
    pub fn settle(&self, at: u32) -> bool {
        at > 0
    }

    pub fn open(amount: u32) -> Charge {
        Charge { amount }
    }
}

pub enum Outcome {
    Settled,
    Declined,
}

pub trait Ledger {
    fn record(&mut self, charge: &Charge);
}

impl Ledger for Charge {
    fn record(&mut self, charge: &Charge) {}
}

pub mod audit {
    pub const LIMIT: u32 = 10;

    pub fn trail() {}

    pub struct Entry;

    impl Entry {
        pub fn at(&self) -> u32 {
            0
        }
    }
}

/* A block comment has nothing after it in this file. */

File: test/fixtures/source/go/sample.go
// A Charge is settled here.
// Each of these lines is a comment of its own.
package charge

// Charge carries what is owed.
type Charge struct {
	Amount uint32
}

// @contract Charge.Settle
func (c *Charge) Settle(at uint32) bool {
	return at > 0
}

func (c Charge) Audit() {}

func Open(amount uint32, notes ...string) *Charge {
	return nil
}

func Grouped(a, b uint32) {}

func Unnamed(uint32) {}

type Ledger interface {
	Record(charge *Charge)
}

const Limit = 10

var Started = false

/* @contract Charge.Audit — a claim in a block comment. */
func Recorded() {}

/* A block comment has nothing after it in this file. */
